apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ElasticApmTracer.java

```diff
@@ -728,6 +731,15 @@ public MetricRegistry getMetricRegistry() {
         return metricRegistry;
     }
 
+    public List<String> getServiceNameOverrides() {
+        List<String> serviceNames = new ArrayList<>(serviceNameByClassLoader.approximateSize());
+        Iterator<Map.Entry<ClassLoader, String>> it = serviceNameByClassLoader.iterator();
```

Comment:
I am not sure if it safe to iterate over the `WeakMap`

Reply:
It is. But I'd recommend iterating using a for-each loop.